### ISSUE TITLE
Icon buttons must have a label for screen readers

### ISSUE THREAD

**Issue by leonvogt** [shoelace-style/webawesome#1542]
Hi there,

Not sure if this counts as a bug, but I’ve noticed that I get a lot of the following warnings in my Svelte app:

> Icon buttons must have a label for screen readers. Add `<wa-icon label="...">` to remove this warning. 
`<wa-button variant="neutral" appearance="accent" size="medium">`

However, my icon already has a label:
```html
<wa-button>
  <wa-icon name="info" label="Info"></wa-icon>
</wa-button>
```

---

This only occurs when using Svelte. With vanilla JS and WebAwesome, the above icon button renders without warnings.
But inside a Svelte 5 app, the icon button renders in the browser as follows:
```html
<wa-button variant="neutral" appearance="accent" size="medium">
   <wa-icon role="img" aria-label="Info" library="default" name="info"></wa-icon>
</wa-button>
```

The issue seems to be that in the rendered version, the icon has an `aria-label` instead of the expected `label` attribute.
I’m not sure at what point in the build process `label` gets transformed into `aria-label`.

WebAwesome specifically checks for `label` [here](https://github.com/shoelace-style/webawesome/blob/0ff5e7fb7a5dd415d488e38d657be9f349883a59/packages/webawesome/src/components/button/button.ts#L184) and prints a warning otherwise.

Question: Would it be possible to enhance that check, and check if either `aria-label` or `label` is present?

Thank you!

---

Freshly created Svelte 5 app with the above icon button and the warning can be found here: 
https://github.com/leonvogt/webawesome-bugreport

**Comment by claviska** [comment]
For some reason the label isn't being rendered, probably because it doesn't reflect and Svelte is setting only the property. However, it's probably easier to check for the existing of the label _property_ which Svelte _is_ setting instead of the label _attribute_, since the property is the source of truth and will always be set.

**Comment by leonvogt** [comment]
Hi @claviska, thanks for the fast reply!
Just to clarify, by checking the property, you mean something like this?
```diff
- if (!hasIconLabel) hasIconLabel = (node as HTMLElement).hasAttribute('label');
+ if (!hasIconLabel) hasIconLabel = (node as HTMLElement).label !== undefined;
```

And yes, you're right! The `label` property is set in the Svelte project, even though only the `aria-label` attribute is shown.

**Comment by claviska** [comment]
Yeah, checking the property like that is more reliable.
